app/react/Viewer/components/OCRDisplay.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { useEffect, useState } from 'react';
+import { connect, ConnectedProps } from 'react-redux';
+import { FileType } from 'shared/types/fileType';
+import { Translate } from 'app/I18N';
+import socket from 'app/socket';
+import { bindActionCreators } from 'redux';
+import { postToOcr, getOcrStatus } from '../actions/ocrActions';
+import { ocrDisplayTips } from '../utils/ocrDisplayTips';
+import { reloadDocument } from '../actions/documentActions';
+
+type OCRDisplayProps = {
+  file: FileType;
+};
+
+const formatDate = (time: number, locale: string) => {
+  const date = new Date(time);
+  return date.toLocaleString(locale);
+};
+
+const mapStateToProps = ({ settings, locale }: any) => {
+  const toggleOCRButton = settings.collection.get('toggleOCRButton');
+  return {
+    ocrIsToggled: toggleOCRButton || false,
+    locale,
+  };
+};
+
+const mapDispatchToProps = (dispatch: any) =>
+  bindActionCreators({ loadDocument: reloadDocument }, dispatch);
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
+type mappedProps = ConnectedProps<typeof connector>;
+type ComponentProps = OCRDisplayProps & mappedProps;
+
+const OCRDisplay = ({ file, ocrIsToggled, locale, loadDocument }: ComponentProps) => {
+  const [ocrStatus, setOcrStatus] = useState({ status: 'loading', lastUpdated: Date.now() });
+
+  const listenOnSuccess = (_id: string) => {
+    if (file._id === _id) {
+      setOcrStatus({ status: 'ocrComplete', lastUpdated: Date.now() });
+      loadDocument(file.entity);
+    }
+  };
+
+  const listenOnError = (_id: string) => {
+    if (file._id === _id) setOcrStatus({ status: 'ocrError', lastUpdated: Date.now() });
+  };
+
+  useEffect(() => {
+    if (ocrIsToggled) {
+      getOcrStatus(file.filename || '')
+        .then(({ status, lastUpdated }) => {
+          setOcrStatus({ status, lastUpdated });
+          if (status === 'inQueue') {
+            // @ts-ignore
+            socket.on('ocr:ready', listenOnSuccess);
+            // @ts-ignore
+            socket.on('ocr:error', listenOnError);
```

Comment:
Did we try to add types for socket? Why are these ignores required?

Reply:
We will fix this by adding the needed parameters to the JS declarations, not by migrating to TS that can take longer.